File: plugin/src/testKit/java/com/mobilesolutionworks/gradle/jacoco/jupiter/TestKitCoverageExtension.java
package com.mobilesolutionworks.gradle.jacoco.jupiter;

import com.mobilesolutionworks.gradle.jacoco.TestKitConfiguration;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.Properties;

public class TestKitCoverageExtension implements GradleRunnerWrapperExtension {

    @Override
    public void configure(GradleRunnerWrapper wrapper) throws GradleRunnerWrapperException {
        try {
            File file = new File(wrapper.getRoot(), "gradle.properties");
            file.createNewFile();

            String jacoco = new TestKitConfiguration("jacoco").agentString;
            Properties properties = new Properties();
            properties.setProperty("org.gradle.jvmargs", jacoco);

            FileOutputStream out = new FileOutputStream(file);
            properties.store(out, "Gradle");
            out.close();
        } catch (IOException e) {
            GradleRunnerWrapperException exception = new GradleRunnerWrapperException();
            exception.initCause(e);

            throw exception;
        }
    }
}

File: plugin/src/testKitKotlin/java/com/mobilesolutionworks/gradle/jacoco/jupiter/TestKitCoverageExtension.kt
package com.mobilesolutionworks.gradle.jacoco.jupiter

import com.mobilesolutionworks.gradle.jacoco.TestKitConfiguration
import java.io.File
import java.util.*


class TestKitCoverageExtension : GradleRunnerWrapperExtension {

    override fun configure(wrapper: GradleRunnerWrapper) {
        File(wrapper.root, "gradle.properties").apply {
            createNewFile()
            TestKitConfiguration("jacoco").agentString?.let {
                val properties = Properties()
                properties.setProperty("org.gradle.jvmargs", it)
                properties.store(outputStream(), "Gradle")
            }
        }
    }
}